### PR TITLE
Fix Null pointer exceptions

### PR DESCRIPTION
**Pull request by res0nance**
Should also prevent https://issues.jenkins.io/browse/JENKINS-64502?focusedCommentId=402814&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-402814

Drops the number of spotbugs to 4 from 10

### CODE REVIEW

**Comment by res0nance** [comment]
CC: @olamy
Probably should add this bit if releasing as this takes care of some edge cases causing null pointers to be produced

File: src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java [hunk 1]
```diff
@@ -133,9 +134,11 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
                     for (Job p : projectList) {
                         //handle non-buildable projects
                         if(!config.canBeScheduled(p)){
+                            User user = User.current();
+                            String userName = user != null ? ModelHyperlinkNote.encodeTo(user) : "unknown";
                             listener.getLogger().println("Skipping " + HyperlinkNote.encodeTo('/'+ p.getUrl(), p.getFullDisplayName()) + 
                                     ". The project is either disabled,"
-                                    + " or the authenticated user " + ModelHyperlinkNote.encodeTo(User.current()) + " has no Item.BUILD permissions,"
+                                    + " or the authenticated user " + userName + " has no Item.BUILD permissions,"
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java:141]: "the authenticated user unknown has no ..." is a bit odd, but should be ok

File: src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java [hunk 2]
```diff
@@ -151,7 +154,10 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
                                     BuildInfoExporterAction.addBuildInfoExporterAction(build, completedRun.getParent().getFullName(), completedRun.getNumber(), completedRun.getResult());
 
                                     if (buildStepResult && config.getBlock().mapBuildStepResult(completedRun.getResult())) {
-                                        build.setResult(config.getBlock().mapBuildResult(completedRun.getResult()));
+                                        Result r = config.getBlock().mapBuildResult(completedRun.getResult());
+                                        if (r != null) {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java:158]: `r == null ` when the run is not finished.
Maybe it makes sense to add a system warning for the edge case

**Review by oleg-nenashev — APPROVED** [review]

File: src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java (hunk 1 above)
Comment (res0nance) [reply src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java:141]: It should be `the authenticated user unknown has no Item.BUILD`

File: src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java (hunk 2 above)
Comment (res0nance) [reply src/main/java/hudson/plugins/parameterizedtrigger/TriggerBuilder.java:158]: Sure I can add it in case there is some underlying issue here

**Pull request merged**